src/components/ViewPhotosDialog/index.tsx

```diff
@@ -68,14 +62,20 @@ export default function ViewPhotosDialog(props: ViewPhotosDialogProps): JSX.Elem
       } else {
         setIsPreviousDisabled(false);
       }
-      setSelectedSlide(myCarousel.current.state.currentSlide);
```

Comment:
This was the primary offender for clients that want to open the modal pointing to a specific slide (specific slide was not honored since this line overrides the specific slide).
Also existing code cannot support opening to a specific slide without using a setTimeout (which is fine for now, we can revisit later).